packages/devtools_app/lib/src/memory/flutter/memory_chart.dart

```diff
@@ -28,64 +27,70 @@ import 'package:mp_chart/mp/core/utils/color_utils.dart';
 import 'package:mp_chart/mp/core/value_formatter/large_value_formatter.dart';
 import 'package:mp_chart/mp/core/value_formatter/value_formatter.dart';
 
+import '../../flutter/auto_dispose_mixin.dart';
+import '../../flutter/controllers.dart';
 import '../../flutter/theme.dart';
-import 'memory_controller.dart';
-
-// TODO(terry): Remove canned data.
-import 'timeseries_data.dart';
+import '../../ui/theme.dart';
+import '../memory_controller.dart';
+import '../memory_protocol.dart';
 
 class MemoryChart extends StatefulWidget {
-  const MemoryChart(this.memoryController);
-
-  final MemoryController memoryController;
-
   @override
   MemoryChartState createState() => MemoryChartState();
 }
 
-class MemoryChartState extends State<MemoryChart> {
+class MemoryChartState extends State<MemoryChart> with AutoDisposeMixin {
   LineChartController _chartController;
 
   LineChartController get chartController => _chartController;
 
+  MemoryController get _controller => Controllers.of(context).memory;
+  
+  MemoryTimeline get _memoryTimeline => _controller.memoryTimeline;
+
+  final legendTypeFace =
+      TypeFace(fontFamily: 'OpenSans', fontWeight: FontWeight.w100);
+
   @override
   void initState() {
     _initController();
 
-    _setupChart();
+    _preloadResources();
 
-    // Read canned data and startup the pseudo-live feed.
-    _initLineData(true);
-
-    // TODO(terry): Remove when live feed is hooked up.
-    // Hookup to replay the canned data.
-    widget.memoryController.addResetFeedListener(() {
-      setState(() {
-        // Reset our starting index into the canned data and
-        // start feeding the live chart.
-        timerDataIndex = 0;
-      });
-    });
+    _setupChart();
 
     super.initState();
   }
 
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    // Process each timeline frame.
+    addAutoDisposeListener(_memoryTimeline.sampleAddedNotifier, () {
+      processLiveData();
+    });
```

Comment:
You can also add this callback as a tearoff, since they're both VoidCallbacks:

```
addAutoDisposeListener(sampleAddedNotifier, processLiveDta);
```

Reply:
Done.